### PR TITLE
Add delete incident comment API

### PR DESCRIPTION
**Pull request by roherzbe**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by roherzbe** [comment]
Hi @leni-msft,

According to the automatic validations it marks that there are 6 Errors & Breaking Changes, all other validations have passed successfully (as far as I can tell).
I looked into the details of these errors and understand the breaking changes were caused by:
•	Adding response call 200 (to an API that returned only 201 beforehand).
•	Adding an Etag field to the resource that is received/sent.

Can you please help me understand why these changes are breaking?
From my understanding and testing if something is added and optional it shouldn’t break previous behavior.

Thanks in advance,
Rotem

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 1]
```diff
@@ -3186,6 +3186,12 @@
           }
         ],
         "responses": {
+          "200": {
```
Comment (pilor) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:3189]: Are these swagger bugfixes to bring swagger in-line with how the API has always behaved?  If not, these should go into a new api-version as the existing preview version is almost 2 years old.

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json (hunk 1 above)
Comment (roherzbe) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:3189]: Yes, this is a bug fix. It exists in production for more then half a year and has no costumer impact (not breaking any scenario). We are aware that in the stable version this kind of change will require a new version.

**Review by pilor — APPROVED** [review]
Swagger fixes for API surface that was already present.  Please work with @leni-msft to deal with any SDK breaking issues and for merging this

**Comment by roherzbe** [comment]
@pilor Thank you very much!
@leni-msft what else needs to be done in order to merge this PR? thanks in advance :)

**Comment by leni-msft** [comment]
@roherzbe please follow the breaking change review process above。

**Comment by roherzbe** [comment]
Hi @leni-msft 
I read the breaking change policy before opening the PR and again now, but I'm not sure in which scenario this PR fits and how to proceed? 
The code I changed is in the preview version. I've also generated a client from this code and validated many scenarios and tests in order to validate this change doesn't break anything.
Should I contact someone for the "BreakingChangeReview" or will someone contact me?
Thanks!

**Comment by leni-msft** [comment]
@roherzbe Adding response code belongs to API behavioral changes. To proceed, please follow the section `3 Detailed Breaking Change Process` in the [breaking change policy](https://aka.ms/AzBreakingChangesPolicy) document.

**Comment by roherzbe** [comment]
@JeffreyRichter 
After discussing this change in office hours and approving this intake (8663535)
Thanks in advance

**Review by JeffreyRichter — APPROVED** [review]

**Comment by roherzbe** [comment]
Jeffrey, Thank you very much!
@leni-msft how do we proceed from here? :)

**Pull request merged**